src/liballoc/rc.rs

```diff
@@ -449,6 +450,8 @@ impl<T: ?Sized> Rc<T> {
     #[stable(feature = "rc_weak", since = "1.4.0")]
     pub fn downgrade(this: &Self) -> Weak<T> {
         this.inc_weak();
+        // Make sure we do not create a dangling Weak
+        debug_assert!(!is_dangling(this.ptr));
```

Comment:
this assert is not gonna show up in libstd for users. Libstd is compiled in release mode

Reply:
We have tests using the debug profile in the CI, so keeping this is fine.

Reply:
Yeah, I didn't want to risk an actual perf hit (though it should get all optimized out, because LLVM knows the pointer to be aligned).